src/commonMain/kotlin/it/krzeminski/snakeyaml/engine/kmp/scanner/StreamReader.kt

```diff
@@ -39,12 +40,20 @@ class StreamReader(
     loadSettings: LoadSettings,
     stream: Source,
 ) {
+    init {
+        require(loadSettings.bufferSize >= 4) { "buffer size must be at least 4 bytes to be able to read all Unicode codepoints" }
+    }
     private val stream: BufferedSource = if (stream is BufferedSource) stream else stream.buffer()
 
     private val name: String = loadSettings.label
 
     private val useMarks: Boolean = loadSettings.useMarks
 
+    /**
+     * Size in bytes what will be used to request a new portion of data from [stream]
+     */
+    private val bufferReadSize: Long = loadSettings.bufferSize.toLong()
```

Comment:
For posterity: right now it's 1024: https://github.com/krzema12/snakeyaml-engine-kmp/blob/cc71a2c8906f6c81ea178fcb141cad60e26c2e64/src/commonMain/kotlin/it/krzeminski/snakeyaml/engine/kmp/api/LoadSettingsBuilder.kt#L42